scipy/stats/_stats.pyx

```diff
@@ -753,3 +753,90 @@ def gaussian_kernel_estimate(points, values, xi, precision, dtype, real _=0):
                 estimate[j, k] += values_[i, k] * arg
 
     return np.asarray(estimate)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def gaussian_kernel_estimate_log(points, values, xi, precision, dtype, real _=0):
+    """
+    def gaussian_kernel_estimate_log(points, real[:, :] values, xi, precision)
+
+    Evaluate the log of the estimated pdf on a provided set of points.
+
+    Parameters
+    ----------
+    points : array_like with shape (n, d)
+        Data points to estimate from in ``d`` dimensions.
+    values : real[:, :] with shape (n, p)
+        Multivariate values associated with the data points.
+    xi : array_like with shape (m, d)
+        Coordinates to evaluate the estimate at in ``d`` dimensions.
+    precision : array_like with shape (d, d)
+        Precision matrix for the Gaussian kernel.
+
+    Returns
+    -------
+    estimate : double[:, :] with shape (m, p)
+        The log of the multivariate Gaussian kernel estimate evaluated at the
+        input coordinates.
+    """
+    cdef:
+        real[:, :] points_, xi_, values_, log_values_
+        real[:, :] whitening, estimate, estimate_, zeros_estimate_
+        real[:] max_, zeros_max
+        int i, j, k
+        int n, d, m, p
+        real arg, residual, log_arg, log_norm
+
+    n = points.shape[0]
+    d = points.shape[1]
+    m = xi.shape[0]
+    p = values.shape[1]
+
+    # Rescale the data
+    whitening = np.linalg.cholesky(precision).astype(dtype, copy=False)
+    points_ = np.dot(points, whitening).astype(dtype, copy=False)
+    xi_ = np.dot(xi, whitening).astype(dtype, copy=False)
+    values_ = values.astype(dtype, copy=False)
+
+    # Log transformation of values_. It is faster than np.log(values_).
+    log_values_ = np.empty((n, p), dtype)
+    for i in range(n):
+        for k in range(p):
+            log_values_[i, k] = math.log(values_[i, k])
+
+    # Evaluate the normalisation
+    log_norm = (- d / 2) * math.log(2 * PI)
+    for i in range(d):
+        log_norm += math.log(whitening[i, i])
```

Comment:
Can this be:
```suggestion
    log_norm += np.diag(whitening).sum()
```
But to follow https://github.com/scipy/scipy/pull/15493/files#r908725848, I suppose I need to figure out how to compute the diagonal elements of the inverse of a matrix from the cholesky decomposition of the matrix. (Could `cho_solve` an identity matrix, but perhaps there is a better way?)